akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -321,6 +322,17 @@ final class Sink[-In, +Mat](delegate: scaladsl.Sink[In, Mat]) extends Graph[Sink
   def mapMaterializedValue[Mat2](f: function.Function[Mat, Mat2]): Sink[In, Mat2] =
     new Sink(delegate.mapMaterializedValue(f.apply _))
 
+  /**
+   * Materializes this Sink, immediately returning (1) its materialized value, and (2) a new Sink
+   * that can be consume elements 'into' the pre-materialized one.
+   *
+   * Useful for when you need a materialized value of a Sink when handing it out to someone to materialize it for you.
+   */
+  def preMaterialize(materializer: Materializer): japi.Pair[Mat @uncheckedVariance, scaladsl.Sink[In @uncheckedVariance, NotUsed]] = {
```

Comment:
scaladsl.Sink in javadsl

Reply:
are these  uncheckedVariance related to https://github.com/akka/akka/pull/24575 ?
add java api compile test?

Reply:
Whoops, thanks